### PR TITLE
[FAB-17350] TC: barebones network test performance

### PR DESCRIPTION
**Pull request by dongmingh**
This CR is to add barebones test on kubernetes.

Signed-off-by: Dongming Hwang <dongming@ibm.com>

### CODE REVIEW

**Comment by lindluni** [comment]
I haven't looked at the code yet, but I did test these files by deploying them to my Kube cluster, everything runs well

File: tools/PTE/CITest/scenarios/FAB-17350.sh [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+# Copyright IBM Corp. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# testcase: FAB-17350
+# 1 channel, 1 org, 1 peer, 1 ca, solo orderer
+# multiple processes (4, 8, 12, ..., 52) X 10000 transactions, both invoke and query
+
+kubeCfg=""
```
Comment (ryjones) [inline tools/PTE/CITest/scenarios/FAB-17350.sh]: `set -ex` somewhere

File: tools/PTE/CITest/scenarios/FAB-17350.sh [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+    exit
+}
+
+networkProc() {
+    cd $OPDir
+    if [ "$kubeCfg" != "" ]; then
+        # bring down network
+        echo "[$0] bring down the kube network"
+        echo "go run main.go -i testdata/barebones-networkspec-spec.yaml -k $kubeCfg -a down"
```
Comment (ryjones) [inline tools/PTE/CITest/scenarios/FAB-17350.sh]: that would obviate these echo statements

**Review by ryjones — CHANGES_REQUESTED** [review]
Please consider adding a `set -ex`, which would dispense with a lot of the echo statements

File: tools/PTE/CITest/scenarios/FAB-17350.sh [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+#operator dir
+OPDir=$FabricTestDir"/tools/operator"
+
+pteReport=$OPDir"/pteReport.txt"
+
+pteInputMaster="testdata/barebones-test-input-master.yml"
+pteInput="testdata/barebones-test-input.yml"
+
+usage () {
+    echo -e "\nUSAGE:\t./FAB-1730.sh [options] [values]"
```
Comment (ryjones) [inline tools/PTE/CITest/scenarios/FAB-17350.sh]: `$0` instead of `./FAB-1730.sh`

File: tools/PTE/CITest/scenarios/FAB-17350.sh [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    echo -e "-c\tbring up network, create/join channel, install/instantiate chaincode"
+    echo -e "\t\t(optional, Default: no)"
+
+    echo -e "-k\tkubenetes configuration file with relative path to fabric-test/tools/operator/"
+    echo -e "\t\ta local network will be brought up if this option is not present"
+
+    echo
+    echo -e "example:"
+    echo -e "\tThe following command will bring up a network on kubenetes based on the specified kube config file, then create/join channel, install/instantiate chaincode, and send transactions."
+    echo -e "\t./FAB-17350.sh -c -k launcher/FAB-17350-kubeConfig/kube-config-wdc04-community-migration.yml"
```
Comment (ryjones) [inline tools/PTE/CITest/scenarios/FAB-17350.sh]: `$0` instead of `./FAB-1730.sh`

File: tools/PTE/CITest/scenarios/FAB-17350.sh (hunk 1 above)
Comment (dongmingh) [reply tools/PTE/CITest/scenarios/FAB-17350.sh]: use -z to check if a variable exists.

File: tools/PTE/CITest/scenarios/FAB-17350.sh (hunk 2 above)
Comment (dongmingh) [reply tools/PTE/CITest/scenarios/FAB-17350.sh]: use -z to check if a variable exist and use set -x instead

File: tools/PTE/CITest/scenarios/FAB-17350.sh (hunk 3 above)
Comment (dongmingh) [reply tools/PTE/CITest/scenarios/FAB-17350.sh]: agree

File: tools/PTE/CITest/scenarios/FAB-17350.sh (hunk 4 above)
Comment (dongmingh) [reply tools/PTE/CITest/scenarios/FAB-17350.sh]: agree

**Review by ryjones — APPROVED** [review]

**Review by lindluni — APPROVED** [review]
LGTM, though the script driven framework for testing is going to be an area of contention moving forward. I know this is how we do it today, but there are going to be vocal voices moving forward about the viability of maintaining all of these scripts

**Pull request merged**